components/support/utils/src/test/java/mozilla/components/support/utils/URLStringUtilsTest.kt

```diff
@@ -67,9 +67,20 @@ class URLStringUtilsTest {
     @Test
     fun isSearchTerm() {
         assertTrue(isSearchTerm("inurl:mozilla.org advanced search"))
-        assertTrue(isSearchTerm("3.14.2019"))
+        assertTrue(isSearchTerm("sf: help"))
+        assertTrue(isSearchTerm("mozilla./~"))
+        assertTrue(isSearchTerm("cnn.com politics"))
 
         assertFalse(isSearchTerm("about:config"))
+        assertFalse(isSearchTerm("about:config:8000"))
         assertFalse(isSearchTerm("http://192.168.255.255"))
+        // Per https://bugs.chromium.org/p/chromium/issues/detail?id=31405, ICANN will accept
+        // purely numeric gTLDs.
+        assertFalse(isSearchTerm("3.14.2019"))
+        assertFalse(isSearchTerm(" cnn.com "))
+        assertFalse(isSearchTerm(" cnn.com"))
+        assertFalse(isSearchTerm("cnn.com "))
+        assertFalse(isSearchTerm("mozilla.com/~userdir"))
+        assertFalse(isSearchTerm("http://faß.de//"))
```

Comment:
Could we include a URL with a unicode character in the TLD?

Reply:
good idea!